src/app/user-page/user-page.component.ts

```diff
@@ -76,5 +82,14 @@ export class UserPageComponent extends Base implements OnInit {
 
   ngOnInit(): void {
     this.getUserInfo(this.username);
+    this.tokenQuery.tokens$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((tokens: TokenUser[]) => {
+      for (const account of this.accountsInfo) {
+        const found = tokens.find((token) => token.tokenSource === account.source);
+        if (found && !['GitHub', 'Google', 'ORCID'].includes(account.name)) {
+          account.username = found?.username;
+          this.otherLinkedAccountsInfo.push(account);
```

Comment:
I am a little wary of modifying the "global" AccountInfo instances with the username. When you leave this page, `account.username` is going to be set to the user. If another component uses the `AccountInfo` instances, the instances will have the left-over username. While it may not matter today in terms of what's displayed in the UI, it could cause some confusion.

I suggest cloning the values, using Object.assign or the spread operator.